lib/commands/context.js

```diff
@@ -1,479 +1,492 @@
+        opts,
+        /** @type {string} */ (/** @type {ADB} */ (this.adb).curDeviceId),
+        /** @type {ADB} */ (this.adb),
+        context
+      );
+      // bind our stop/exit handler, passing in context so we know which
+      // one stopped unexpectedly
+      cd.on(Chromedriver.EVENT_CHANGED, (msg) => {
+        if (msg.state === Chromedriver.STATE_STOPPED) {
+          this.onChromedriverStop(context);
+        }
+      });
+      // save the chromedriver object under the context
+      this.sessionChromedrivers[context] = cd;
+    }
+    // hook up the local variables so we can proxy this biz
+    this.chromedriver = cd;
+    this.proxyReqRes = this.chromedriver.proxyReq.bind(this.chromedriver);
+    this.proxyCommand = /** @type {import('@appium/types').ExternalDriver['proxyCommand']} */ (
+      this.chromedriver.jwproxy.command.bind(this.chromedriver.jwproxy)
+    );
+    this.jwpProxyActive = true;
+  },
+
+  // Stop proxying to any Chromedriver
+  suspendChromedriverProxy() {
+    this.chromedriver = null;
+    this.proxyReqRes = undefined;
+    this.proxyCommand = undefined;
+    this.jwpProxyActive = false;
+  },
+
+  // Handle an out-of-band Chromedriver stop event
+  async onChromedriverStop(context) {
+    this.log.warn(`Chromedriver for context ${context} stopped unexpectedly`);
+    if (context === this.curContext) {
+      // we exited unexpectedly while automating the current context and so want
+      // to shut down the session and respond with an error
+      let err = new Error('Chromedriver quit unexpectedly during session');
+      await this.startUnexpectedShutdown(err);
+    } else {
+      // if a Chromedriver in the non-active context barfs, we don't really
+      // care, we'll just make a new one next time we need the context.
+      this.log.warn(
+        "Chromedriver quit unexpectedly, but it wasn't the active " + 'context, ignoring'
+      );
+      delete this.sessionChromedrivers[context];
+    }
+  },
+
+  // Intentionally stop all the chromedrivers currently active, and ignore
+  // their exit events
+  async stopChromedriverProxies() {
+    this.suspendChromedriverProxy(); // make sure we turn off the proxy flag
+    for (let context of _.keys(this.sessionChromedrivers)) {
+      let cd = this.sessionChromedrivers[context];
+      this.log.debug(`Stopping chromedriver for context ${context}`);
+      // stop listening for the stopped state event
+      cd.removeAllListeners(Chromedriver.EVENT_CHANGED);
+      try {
+        await cd.stop();
+      } catch (err) {
+        this.log.warn(`Error stopping Chromedriver: ${/** @type {Error} */ (err).message}`);
+      }
+      delete this.sessionChromedrivers[context];
+    }
+  },
+
+  isChromedriverContext(viewName) {
+    return _.includes(viewName, WEBVIEW_WIN) || viewName === CHROMIUM_WIN;
+  },
+
+  shouldDismissChromeWelcome() {
+    return (
+      !!this.opts.chromeOptions &&
+      _.isArray(this.opts.chromeOptions.args) &&
```

Comment:
should we suppress the warning below?

Reply:
I don't see a warning?

Reply:
I assume it is a beta github feature.
It complains about `args` property is unknown for `chromeOptions`

Reply:
oh..  yeah.  as I said, it won't work until we get those updates in.